mmv1/products/dataproc/Batch.yaml

```
@@ -0,0 +1,504 @@
+    test_env_vars:
+      project_name: :PROJECT_NAME
+    vars:
+      subnetwork_name: 'default'
+      prevent_destroy: 'true'
+    test_vars_overrides:
+      subnetwork_name: 'acctest.BootstrapSubnetWithFirewallForDataprocBatches(t, "dataproc-pyspark-test-network", "dataproc-pyspark-test-subnetwork")'
+      prevent_destroy: 'false'
+    skip_import_test: true
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dataproc_batch_sparkr'
+    primary_resource_id: 'example_batch_sparkr'
+    primary_resource_name:
+      'fmt.Sprintf("tf-test-spark-batch%s", context["random_suffix"])'
+    test_env_vars:
+      project_name: :PROJECT_NAME
+    vars:
+      subnetwork_name: 'default'
+      prevent_destroy: 'true'
+    test_vars_overrides:
+      subnetwork_name: 'acctest.BootstrapSubnetWithFirewallForDataprocBatches(t, "dataproc-pyspark-test-network", "dataproc-pyspark-test-subnetwork")'
+      prevent_destroy: 'false'
+    skip_import_test: true
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  decoder: templates/terraform/decoders/cloud_dataproc_batch.go.erb
+  constants: templates/terraform/constants/cloud_dataproc_batch.go.erb
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    url_param_only: true
+    immutable: true
+    description: |
+      The location in which the batch will be created in.
+  - !ruby/object:Api::Type::String
+    name: 'batchId'
+    url_param_only: true
+    immutable: true
+    description: |
+      The batch Id of the batch.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The resource name of the batch.
+  - !ruby/object:Api::Type::String
+    name: 'uuid'
+    output: true
+    description: |
+      A batch UUID (Unique Universal Identifier). The service generates this value when it creates the batch.
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    output: true
+    description: |
+      The time when the batch was created.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'runtimeInfo'
+    description: 'Runtime information about batch execution.'
+    output: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'outputUri'
+        description: |
+          A URI pointing to the location of the stdout and stderr of the workload.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'diagnosticOutputUri'
+        description: |
+          A URI pointing to the location of the diagnostics tarball.
+        output: true
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'endpoints'
+        description: |
+          Map of remote access endpoints (such as web interfaces and APIs) to their URIs.
+        output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'approximateUsage'
+        description: |
+          Approximate workload resource usage, calculated when the workload completes.
```

This could be useful to include. There's also a note in the [API docs](https://cloud.google.com/dataproc-serverless/docs/reference/rest/v1/RuntimeInfo) about the meaning of this changing in the future; not sure if you'd want to include that here as well?

I don't think duplicating the pricing link on the subfields is necessary, personally.

```suggestion
          Approximate workload resource usage, calculated when the workload completes (see [Dataproc Serverless pricing](https://cloud.google.com/dataproc-serverless/pricing)).
```